src/dr/app/beast/BeastMain.java

```diff
@@ -611,6 +612,20 @@ public static void main(String[] args) throws java.io.IOException {
             }
         }
 
+        if (arguments.hasOption("fail_threads")) {
+
+            Thread.UncaughtExceptionHandler handler = new Thread.UncaughtExceptionHandler() {
+                @Override
+                public void uncaughtException(Thread t, Throwable e) {
+                    System.err.println("Error in thread " + t.getName() + ": ");
+                    e.printStackTrace();
+                    System.exit(1);
```

Comment:
1. Should return `-1` 
2. fix typo in line 339.

Reply:
@msuchard Done.